### PR TITLE
Moved Coding Logic into separate framework

### PR DESCRIPTION
**Pull request by jay18001**
The encoding and decoding (coding) logic doesn't have any references on the reset so it makes sense for them to be two separate frameworks.

- Moved all the source files for both targets into `Sources/Kumo`  and `Sources/KumoCoding`
- Moved the tests `Tests/KumoTests` (It depends on both frameworks)
- Updated the Package managers targets, Swift and Cocoapods
- Add a new target to the Xcode project
- Added Mac and TVOS targets
- Updated gitignore

### CODE REVIEW

File: Sources/KumoCoding/KumoCoding.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+//
```
Comment (powerje) [inline Sources/KumoCoding/KumoCoding.h:1]: Please delete the header entirely

**Comment by powerje** [comment]
Thanks for contributing!

Apologies, we haven't set CI up for this project yet.  There is a build error in the Kumo target:

`error: Build input file cannot be found: '.../Kumo/Sources/Kumo/Extensions/StringEncoding.swift' (in target 'Kumo' from project 'Kumo')`

Could you fix this issue?

File: Sources/Kumo/Errors/ResponseError.swift [hunk 2]
```diff
@@ -1,5 +1,9 @@
 import Foundation
 
+#if canImport(KumoCoding)
```
Comment (powerje) [inline Sources/Kumo/Errors/ResponseError.swift:3]: Is the `canImport` necessary? I think with this change Kumo depends entirely on KumoCoding so it should always be importable?

Comment (jay18001) [reply Sources/Kumo/Errors/ResponseError.swift:3]: Yes because cocoapods don’t allow multiple frameworks in a single podfiles, As far as I can tell.

Comment (powerje) [reply Sources/Kumo/Errors/ResponseError.swift:3]: Ah I see, that makes sense.

I think in that case we would need to create a separate podspec for KumoCoding, similar to how RxSwift is setup with RxCocoa:

https://github.com/ReactiveX/RxSwift/blob/master/RxSwift.podspec
https://github.com/ReactiveX/RxSwift/blob/master/RxCocoa.podspec

That said, we can handle that in a future PR

File: Sources/KumoCoding/KumoCoding.h (hunk 1 above)
Comment (powerje) [reply Sources/KumoCoding/KumoCoding.h]: Apologies! I meant the file header comments, I believe the header file itself is necessary 😅

**Comment by powerje** [comment]
@jay18001 thanks so much for your contribution!

**Pull request merged**